operator/rpc_client.go

```diff
@@ -2,421 +2,259 @@ package operator
 
 import (
 	"errors"
-	"fmt"
-	"net"
 	"net/rpc"
-	"sync"
 	"time"
 
 	"github.com/Layr-Labs/eigensdk-go/logging"
 	eigentypes "github.com/Layr-Labs/eigensdk-go/types"
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/prometheus/client_golang/prometheus"
-
 	"github.com/NethermindEth/near-sffl/core"
 	"github.com/NethermindEth/near-sffl/core/types/messages"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
-const (
-	ResendInterval = 2 * time.Second
-	MaxRetries     = 10
-)
-
-type AggregatorRpcClienter interface {
-	core.Metricable
-
-	SendSignedCheckpointTaskResponseToAggregator(signedCheckpointTaskResponse *messages.SignedCheckpointTaskResponse)
-	SendSignedStateRootUpdateToAggregator(signedStateRootUpdateMessage *messages.SignedStateRootUpdateMessage)
-	SendSignedOperatorSetUpdateToAggregator(signedOperatorSetUpdateMessage *messages.SignedOperatorSetUpdateMessage)
-	GetAggregatedCheckpointMessages(fromTimestamp, toTimestamp uint64) (*messages.CheckpointMessages, error)
-}
-
-type unsentRpcMessage struct {
-	Message interface{}
-	Retries int
-}
-
-type AggregatorRpcClient struct {
-	rpcClientLock              sync.RWMutex
-	rpcClient                  *rpc.Client
-	aggregatorIpPortAddr       string
-	registryCoordinatorAddress common.Address
-
-	operatorId          eigentypes.OperatorId
-	notifiedInitialized bool
-
-	unsentMessagesLock sync.Mutex
-	unsentMessages     []unsentRpcMessage
-	resendTicker       *time.Ticker
-
-	logger   logging.Logger
-	listener RpcClientEventListener
+type RpcClient interface {
+	Call(serviceMethod string, args any, reply any) error
+	// TODO: Do we also want a `Close` method?
```

Comment:
I'm against the idea of adding a `Close` method here, but it would allow us to add a `Close` method to the `AggregatorRpcClient` such that it would close the provided `RpcClient`. Ideally, a component should not be responsible for closing the provided dependencies, but it could be a quick & dirty solution.